From d89e2607077f96467f75971e5efb79b589ef073d Mon Sep 17 00:00:00 2001
Message-Id: <patch-5.5-d89e2607077-20220901T130817Z-avarab@gmail.com>
In-Reply-To: <cover-0.5-00000000000-20220901T130817Z-avarab@gmail.com>
References: <sso99so6-n28s-rq86-8q20-4456r3pn869r@tzk.qr>
	<cover-0.5-00000000000-20220901T130817Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 1 Sep 2022 13:57:36 +0200
Subject: [PATCH 5/5] Makefile: fix "make clean && make bin-wrappers/$NAME"
 dependencies
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The "bin-wrappers/" scripts have never had properly declared
dependency information, i.e. this has never worked:

	make clean &&
	make bin-wrappers/git &&
	./bin-wrappers/git

There was no reason not to have that work, just as most things
generated by the Makefile have proper dependency information.

Before the preceding commits doing this would have been painful, but
now it's quite easy to simply make use of the third parameter to our
"bin_wrappers_template"

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 533858ca481..07a9901c99d 100644
--- a/Makefile
+++ b/Makefile
@@ -3075,9 +3075,10 @@ define bin_wrappers_template
 BW_$(1) = $$($(1):%=bin-wrappers/%)
 BIN_WRAPPERS += $$(BW_$(1))
 all:: $$(BW_$(1))
+$$(BW_$(1)): bin-wrappers/% : $(3)%$(4)
 $$(BW_$(1)): wrap-for-bin.sh
 	$$(call mkdir_p_parent_template)
-	$$(QUIET_GEN)$$(call cmd_munge_bin_wrappers_script,$(2),$(3))
+	$$(QUIET_GEN)$$(call cmd_munge_bin_wrappers_script,$(2),$(3),$(4))
 endef
 
 $(eval $(call bin_wrappers_template,BINDIR_PROGRAMS_NEED_X,'$$(@F)',,$$X))
-- 
2.37.3.1426.g360dd7cf8ca

